spec/api_auth_spec.rb

```diff
@@ -94,6 +112,24 @@ def hmac(secret_key, request)
       request['date'] = "٢٠١٤-٠٩-٠٨ ١٦:٣١:١٤ +٠٣٠٠"
       expect(ApiAuth.authentic?(request, "123")).to eq false
     end
+
+    context "canonical string contains the http_method" do
```

Comment:
Should this context maybe contain a test that checks for a request where the method doesn't match the one it was signed with?